### PR TITLE
Fix TODO, style and naming

### PR DESCRIPTION
**Pull request by hgranule**
* Fix styleguide and naming
* Add GB_test.h
* Remove membus impl
* Add few scripts
* Some useless funcs are deleted
* Add new constructors test

### CODE REVIEW

File: include/device/GB_joypad.h [hunk 1]
```diff
@@ -49,17 +45,18 @@ class JoyPad {
...
+    void __raise_interrupt();
+
  public:
 
     explicit
-    JoyPad(InterruptController* ic_link)  : __interrupt_link(ic_link)
+    JoyPad(InterruptController* ic_link = nullptr)  : __interrupt_link(ic_link)
                                         , __previous_step_key_set(0)
                                         , __pressed_key_set(0)
                                         , __p14(true)
```
Comment (dolovnyak) [inline include/device/GB_joypad.h:57]: Maybe also use init values from GB_config for p14, p15?

File: sources/interrupt.cc [hunk 2]
```diff
@@ -3,29 +3,28 @@
 namespace GB::device {
 
     using IC    = InterruptController;
-    using Mbus  = ::GB::memory::BusInterface;
 
     // Memory mapping
 
-    static void write_IF_reg(IC* ic, word_t, byte_t val) {
```
Comment (dolovnyak) [inline sources/interrupt.cc:10]: suggest add TODO:
When membus will implemented delete or implement commented functions.

File: sources/joypad.cc [hunk 3]
```diff
@@ -1,22 +1,21 @@
 #include "device/GB_joypad.h"
-
 #include "memory/GB_vaddr.h"
 
 namespace GB::device {
 
-    static byte_t read_P1_reg(JoyPad* jp, word_t) {
```
Comment (dolovnyak) [inline sources/joypad.cc:7]: Also TODO

File: sources/wram.cc [hunk 4]
```diff
@@ -1,87 +1,49 @@
 #include "device/GB_wram.h"
-
 #include "memory/GB_vaddr.h"
 
-
 namespace GB::device {
 
-WRAM::WRAM()
```
Comment (dolovnyak) [inline sources/wram.cc:8]: Also TODO

File: utils/run_all_tests [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    printf "${COLOR_SET}${HEADER}\n| Running test:  %$((${#HEADER} - 19))s |\n${HEADER}${COLOR_RESET}\n" ${UNIT_TEST}
+    if !(exec ${unit_test_exe} $@);
+    then
+        printf "Test failed, exiting ..."
+        exit 1;
+    fi
+done
+
+printf "\n\n"
```
Comment (dolovnyak) [inline utils/run_all_tests:30]: no new line before eof

Comment (hgranule) [reply utils/run_all_tests:30]: I made it there, because it is simetric!
```

AAA

AAA

AAA

```

File: sources/wram.cc (hunk 4 above)
Comment (hgranule) [reply sources/wram.cc:8]: TODO WHAT?

Comment (hgranule) [reply sources/wram.cc:8]: It just deleted forever from CC file

File: sources/interrupt.cc (hunk 2 above)
Comment (hgranule) [reply sources/interrupt.cc:10]: I need to leave that code for my future work

File: include/device/GB_joypad.h (hunk 1 above)
Comment (hgranule) [reply include/device/GB_joypad.h:57]: It isn't registers, so I used it.
It actually has no initial values at all, but we need something normal here

File: utils/run_all_tests (hunk 5 above)
Comment (hgranule) [reply utils/run_all_tests:30]: It isn't google c++ styleguide goal. It is SHELL FILE, I think it's not necessary here.

File: sources/wram.cc (hunk 4 above)
Comment (hgranule) [reply sources/wram.cc:8]: I made an implementation of it in header file.

File: sources/joypad.cc (hunk 3 above)
Comment (hgranule) [reply sources/joypad.cc:7]: I need to leave that code for my future work

File: sources/interrupt.cc (hunk 2 above)
Comment (dolovnyak) [reply sources/interrupt.cc:10]: I suggest add TODO for all commented code, not delete it

File: include/device/GB_joypad.h (hunk 1 above)
Comment (hgranule) [reply include/device/GB_joypad.h:57]: p14 and p15 are input lines, so there are no initial values for them. P1 is a fake register, that implemented as a register in GameBoy virtual memory space.

File: sources/interrupt.cc (hunk 2 above)
Comment (hgranule) [reply sources/interrupt.cc:10]: I already have an TODO for unit-tests. There are no sense at all if I have TODO for each commented function, or commented file.

File: sources/joypad.cc (hunk 3 above)
Comment (dolovnyak) [reply sources/joypad.cc:7]: I mean add TODO in each file before commented code, for everyone to know what there things are needed

File: sources/wram.cc (hunk 4 above)
Comment (dolovnyak) [reply sources/wram.cc:8]: I mean add TODO in each file before commented code, for everyone to know what there things are needed

File: sources/joypad.cc (hunk 3 above)
Comment (hgranule) [reply sources/joypad.cc:7]: Answered in previous comment!

**Review by dolovnyak — APPROVED** [review]

**Pull request merged**